### ISSUE TITLE
Cluster Templates Create with Yaml 

### ISSUE THREAD

**Issue by westlywright** [rancher/rancher#21579]
When creating cluster templates the creator should be able to use the yaml editor as they would when launching a stand alone cluster. When switching to yaml mode the user will loose the ability easily toggle overrides and will have to manually add overrides at the bottom of the form. The caveat to this is that if the user selected a bunch of overrides previous to selecting yaml mode those overrides would persist in the overrides section. When a user switches from edit as yaml to the edit form they should be presented with a warning about loosing fields that are not present in the UI (this exists for another edit as yaml form already) and given the opportunity to cancel.

When a user is launching a cluster from a template that was created from yaml the user will be presented to with the read only form except for fields where an override exists, the same as it functions currently. Any additional params the template creator provided via the yaml will only be viewable in a read-only yaml view unless a corresponding override exists, then it will be in the overrides section.

**Comment by davidnuzik** [comment]
Version: master-head (7/30/19)

I am reopening the issue. It looks like the "Edit as YAML" link is not shown when we edit a cluster created with a cluster template.

It may also be worth noting that even if we create a cluster without a cluster template, edit the cluster, the "Edit as YAML" link is also not shown here as well. For that reference: https://github.com/rancher/rancher/issues/21866

**Comment by westlywright** [comment]
Edit as Yaml is visible when editing a cluster not launched with a template now.

**Comment by soumyalj** [comment]
Tested with master-head build. Use cases tested :
1. a.  As a global admin, create a cluster template T1 and revision R1 where few fields are provided in "Edit as yaml"
```
    kubelet:
      extra_args:
        authorization-mode: "AlwaysAllow"
        application-metrics-count-limit: 150
```
b. Create a cluster with R1
c. Verify kubelet parameters in docker inspect
2.  a.  As a global admin, create a cluster template T1 and revision R1 where few fields in UI are added as overrides. Also have few fields are provided in "Edit as yaml"
```
    kubelet:
      extra_args:
        authorization-mode: "AlwaysAllow"
        application-metrics-count-limit: 150
```
b. Create a cluster with R1
c. Verify kubelet parameters in docker inspect

Related issues: https://github.com/rancher/rancher/issues/21928

**Comment by davidnuzik** [comment]
Version: master-head (8/6/19)

Confirmed the "Edit as YAML" option is back in when editing a cluster.

I also executed tests outlined by Soumya above.
This includes situations where we have overrides for this via `rancherKubernetesEngineConfig.services.kubelet.extraArgs.authorization-mode` for example.

I also ensured all of the essentials Wes pointed out in the issue description are true:

- Able to set overrides in form and then switch to YAML and keep those overrides.
- Able to switch back to edit as form - note an error is no longer shown -- settings are kept (New change, expected). If any settings are invalid they are dropped when create CT. (This is expected).
- Any overrides provided for custom options (such as kubelet ExtraArgs) can indeed be overriden in YAML when leveraging the CT. If an override is NOT provided then they CANNOT be overriden.

All these areas were validated and I did not identify any unexpected (not yet known) issues.
